aea/cli/common.py

```diff
@@ -496,3 +507,77 @@ def _find_item_locally(ctx, item_type, item_public_id) -> Path:
         sys.exit(1)
 
     return package_path
+
+
+def _check_package_fingerprints(
+    package_configuration: PackageConfiguration, package_directory: Path
+):
+    """Given a package configuration and a package directory, check whether
+    the hash of the files in the package directory match the one in the package configurations."""
+    hasher = IPFSHashOnly()
+    expected_fingerprints = package_configuration.fingerprint
+
+    # compute fingerprints
+    actual_fingerprints = {}  # type: Dict[str, str]
+    # find all valid files of the package
+    all_files = [
+        x
+        for x in package_directory.glob("**/*")
+        if x.is_file()
+        and not x.match("*__pycache__/*")
+        and not x.match(
+            "*.yaml"
+        )  # TODO this should only match "protocol.yaml", "connection.yaml" etc.
+    ]
+
+    for file in all_files:
+        file_hash = hasher.get(str(file))
+        key = str(file.relative_to(package_directory))
+        assert key not in actual_fingerprints  # nosec
+        actual_fingerprints[key] = file_hash
+
+    if expected_fingerprints != actual_fingerprints:
+        raise ValueError(
+            "Fingerprints for package {} do not match:\nExpected: {}\nActual: {}".format(
+                package_directory,
+                pprint.pformat(expected_fingerprints),
+                pprint.pformat(actual_fingerprints),
+            )
+        )
+
+
+def _validate_config_consistency(ctx: Context):
+    """
+    Validate fingerprints for every agent component.
+
+    :raise ValueError: if the fingerprints do not match
+                       or if the packages are bad formatted (e.g. missing configuration file.)
+    """
+
+    packages_public_ids_to_types = dict(
+        [
+            *map(lambda x: (x, ConfigurationType.PROTOCOL), ctx.agent_config.protocols),
+            *map(
+                lambda x: (x, ConfigurationType.CONNECTION),
+                ctx.agent_config.connections,
+            ),
+            *map(lambda x: (x, ConfigurationType.SKILL), ctx.agent_config.skills),
+            *map(lambda x: (x, ConfigurationType.CONTRACT), ctx.agent_config.contracts),
+        ]
+    )  # type: Dict[PublicId, ConfigurationType]
+
+    for public_id, item_type in packages_public_ids_to_types.items():
+        if public_id.author == ctx.agent_config.public_id.author:
```

Comment:
It is worth checking, but I think if one adds a package from the registry it always ends up in vendors, even if it is of the same author. This makes sense as vendors contains all finished packages even self authored ones.

Reply:
see if the provided solution works. We give precedence to custom packages.